internal/crd/crd.go

```diff
@@ -63,6 +64,38 @@ func Validate(ctx context.Context, cl client.Client, newCrd *apiextensionsv1.Cus
 	return nil
 }
 
+// CreateOrUpdateCRD performs all the necessary actions to either Create or Update a CRD,
+// including running validation when an Update is required.
+func CreateOrUpdateCRD(ctx context.Context, cl client.Client, newCrd *apiextensionsv1.CustomResourceDefinition) error {
+	// First try to create the CRD
```

Comment:
What's the code path for the case where the CRD already exists and is identical to `newCRD`? It seems like we try (and fail) to `Create`, and then we `Update`, with the end result being a no-op.

Do you think we can:
1. `Get`
2. If not exist, `Create`
3. If exist, compare
4. If diff, update.

That way we don't hit the apiserver at all when there's nothing to do.

Reply:
Yeah, that's a good question. I think we generally _always_ want to take _exactly_ what's in `new` and essentially erase stuff in `old` that isn't in `new`. 

But you're right that there are exceptions. Off the top of my head, I'd exclude:
- everything in `metadata` except name, namespace, labels, and annotations.
- all of `status`

Does `last-applied-configuration` show up in rukpak-managed CRDs right now?